modules/blockbuilder/config.go

```diff
@@ -38,13 +39,26 @@ type Config struct {
 
 	LookbackOnNoCommit time.Duration `yaml:"lookback_on_no_commit" category:"advanced"`
 
-	blockConfig BlockConfig `yaml:"block" doc:"Configuration for the block builder."`
+	BlockConfig BlockConfig `yaml:"block" doc:"Configuration for the block builder."`
+	WAL         wal.Config  `yaml:"wal" doc:"Configuration for the write ahead log."`
 
 	// This config is dynamically injected because defined outside the ingester config.
 	IngestStorageConfig ingest.Config `yaml:"-"`
 }
 
 func (c *Config) Validate() error {
+	if c.BlockConfig.BlockCfg.Version != c.WAL.Version {
```

Comment:
If the WAL and block versions are required to be the same, then maybe we should swap the validation logic to prevent the WAL version from being set, and always copy over BlockCfg.Version.

Reply:
Good point, I'll change it to that.